source/fundamentals/data-formats/documents.txt

```diff
@@ -14,34 +14,379 @@ Documents
+  This means that if the interface was ever changed in a way that broke
+  the binary compatibility, all applications and classes that use the
+  interface need to be recompiled to run the new version without errors.
+
+In the following code snippet, we show how to instantiate and build a sample
+``BasicDBObject`` instance representing a document containing several
+different field types:
+
+.. code-block:: java
+
+   BasicDBObject author = new BasicDBObject("_id", new ObjectId())
+       .append("name", "Gabriel García Márquez")
+       .append("lastUpdated", new Date(120, 2, 26))
+       .append("novels", Arrays.asList(
+           new BasicDBObject("title", "One Hundred Years of Solitude").append("yearPublished", 1967),
+           new BasicDBObject("title", "Chronicle of a Death Foretold").append("yearPublished", 1981),
+           new BasicDBObject("title", "Love in the Time of Cholera").append("yearPublished", 1985)));
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection()</apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``BasicDBObject`` class. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<BasicDBObject> collection = database.getCollection("authors", BasicDBObject.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample document
+data from the collection using the following code:
+
+.. code-block:: java
+
+   import com.mongodb.client.model.Filters;
+   // <MongoCollection setup code here>
+
+   BasicDBObject doc = collection.find(Filters.eq("name", "Gabriel García Márquez")).first();
+   if (doc != null) {
+       System.out.println("_id: " + doc.getObjectId("_id")
+           + ", name: " + doc.getString("name")
+           + ", lastUpdated: " + doc.getDate("lastUpdated"));
+
+       BasicDBList novels = (BasicDBList) doc.get("novels");
+       if (novels != null) {
+           BasicDBObject[] novelArr = novels.toArray(new BasicDBObject[0]);
+           for (BasicDBObject novel : novelArr) {
+               System.out.println("title: " + novel.getString("title")
+               + ", yearPublished: " + novel.getInt("yearPublished"));
+           }
+       }
+   }
+
+.. note::
+
+   The code sample above uses helper methods that check the returned type
+   and throw an exception if it is unable to cast the field value.
+   You can call the :java-docs:`get() </apidocs/bson/org/bson/BasicBSONObject.html?is-external=true#get(java.lang.String)>`
+   method to retrieve values as type type ``Object`` and to skip type checking.
+
+The output should resemble the following:
```

Comment:
"will" instead of "should"?

Reply:
I avoid using "will" across docs because it often leads to passive voice and I think we prefer active when possible based on some feedback I received last year. Do you think that I should change it anyway?